pkg/controller/nodeipam/ipam/multicidrset/multi_cidr_set.go

```diff
@@ -359,3 +363,7 @@ func (s *MultiCIDRSet) getIndexForIP(ip net.IP) (int, error) {
 func (s *MultiCIDRSet) UpdateEvaluatedCount(evaluated int) {
 	cidrSetAllocationTriesPerRequest.WithLabelValues(s.Label).Observe(float64(evaluated))
 }
+
+func getMaxCIDRs(subNetMaskSize, clusterMaskSize int) int {
+	return 1 << uint32(subNetMaskSize-clusterMaskSize)
```

Comment:
validate the values and return errors, actually, I think we should do something better, and include this logic too,

```
	clusterMask := cidrConfig.Mask
	clusterMaskSize, bits := clusterMask.Size()
	var subNetMaskSize int
	switch /*v4 or v6*/ {
	case netutils.IsIPv4(cidrConfig.IP):
		subNetMaskSize = 32 - perNodeHostBits
	case netutils.IsIPv6(cidrConfig.IP):
		subNetMaskSize = 128 - perNodeHostBits
	}
	if netutils.IsIPv6(cidrConfig.IP) && (subNetMaskSize-clusterMaskSize > clusterSubnetMaxDiff) {
		return nil, &CIDRSetSubNetTooBigErr{
			cidr:            cidrConfig.String(),
			subnetMaskSize:  subNetMaskSize,
			clusterMaskSize: clusterMaskSize,
		}
	}

```

so the function will be 
```go
// getMaxCIDRs returns that number of CIDRs that can be obtained subdividing the network on
// subnets with the give size 
func getMaxCIDRs(network *net.IPNet, subnetMask int) (int,error)
```


```

but I don't want to push that on you, I'm fine if you decide to postpone